repos/system_upgrade/common/actors/reportleftoverpackages/reportleftoverpackages/libraries/reportleftoverpackages.py

```diff
@@ -0,0 +1,60 @@
+from leapp import reporting
+from leapp.libraries.stdlib import api
+from leapp.models import LeftoverPackages, RemovedPackages
+
+FMT_LIST_SEPARATOR = '\n    - '
+
+
+def process():
+    removed_packages = next(api.consume(RemovedPackages), None)
+    leftover_packages = next(api.consume(LeftoverPackages), LeftoverPackages())
+    leftover_pkgs_to_remove = ['-'.join([pkg.name, pkg.version, pkg.release]) for pkg in leftover_packages.items]
+
+    if removed_packages:
+        title = 'Leftover RHEL packages have been removed'
+
+        if removed_packages.items:
+            removed = ['-'.join([pkg.name, pkg.version, pkg.release]) for pkg in removed_packages.items]
+            reporting.create_report([
+                reporting.Title(title),
+                reporting.Summary('Following packages have been removed:\n{}'.format('\n'.join(removed))),
+                reporting.Severity(reporting.Severity.HIGH),
+                reporting.Groups([reporting.Groups.SANITY]),
+            ] + [reporting.RelatedResource('package', pkg.name) for pkg in removed_packages.items])
+        else:
+            summary = (
+                'Following packages have been removed:{sep}{list}\n'
+                'Dependent packages may have been removed as well, please check that you are not missing '
+                'any packages.'
+                .format(
+                    sep=FMT_LIST_SEPARATOR,
+                    list=FMT_LIST_SEPARATOR.join(leftover_pkgs_to_remove)
+                )
+            )
+
+            reporting.create_report([
+                reporting.Title(title),
+                reporting.Summary(summary),
+                reporting.Severity(reporting.Severity.HIGH),
+                reporting.Groups([reporting.Groups.SANITY]),
+            ] + [reporting.RelatedResource('package', pkg.name) for pkg in leftover_packages.items])
```

Comment:
There is a bug introduced in the initial implementation, this report should've reported that leapp failed to remove those packages. (as the actor docs state)

Reply:
Refactored the code so it should be now more suitable the purpose.